main.py

```diff
@@ -393,8 +412,16 @@ async def inner_f():
             embed = Embed(title=embed_title, description=embed_content)
             await current_channel.send(embed=embed)
 
+            bust_num = current_bust_invocation
             await asyncio.sleep(seconds_between_songs)
 
+            # We ensure the current bust invocation is the same as
+            # when we started, since it is possible we stopped this bust
+            # and started a new one while sleeping
+            if current_channel_content is None or current_bust_invocation != bust_num:
+                # We ran !stop while sleeping
+                return
```

Comment:
Another way we could implement this is to simply cancel the execution of this whole function when `!stop` is called.

We can wrap this coroutine (`inner_f`) in an [asyncio.Task](https://docs.python.org/3/library/asyncio-task.html#task-object), which can be interrupted whenever we like. Thus we don't need to worry about this coroutine continuing on with false assumptions.

This commit demonstrates how it might be done: fd1c754

Reply:
I'd like to ask a few questions as I'm not super familiar with how coroutines work:

1) Since we call `finish_bust()` from within `inner_f()`, and `finish_bust()` cancels the `play_next_song_task` (which is running `inner_f()`), is there any danger of `finish_bust()` cancelling itself? this doesn't seem to happen in my testing, but with async code that doesn't mean it can't happen

2) When implementing the changes in your test commit I get an inconsistently reproducable error
```
Task exception was never retrieved
future: <Task finished name='Task-227' coro=<play_next_song.<locals>.inner_f() done, defined at /home/jack/Unsynced/busty/main.py:389> exception=TypeError("'NoneType' object is not subscriptable")>
Traceback (most recent call last):
  File "/home/jack/Unsynced/busty/main.py", line 414, in inner_f
    current_channel_content = current_channel_content[skip_count:]
```
I've been able to reproduce it at least 3x by 1) starting a bust, 2) stopping during the first song, 3) starting a bust again (though it doesn't happen every time). The exception happens right when the first "Currently chillin" pops up. The bust itself appears normal from a user's perspective, so I assume this is somehow a leftover from the previous bust which is throwing an exception.

I'm pretty sure I implemented all your suggestions correctly, but just in case I've pushed the code that throws this exception so you can see it yourself.